lib/task-data/schemas/generate-tag.json

```diff
@@ -0,0 +1,41 @@
+{
+    "$schema": "rackhd-task-schema.json",
+    "copyright": "Copyright 2016, EMC, Inc.",
+    "title": "Generate Tag Task",
+    "description": "The schema for generate-tag",
+    "describeJob": "Job.Catalog.GenerateTag",
+    "allOf": [
+        { "$ref": "common-task-options.json#/definitions/Options" },
+        {
+            "oneOf": [
+                {
+                    "properties": {
+                        "nodeId": {
+                            "type": "string",
+                            "description": "An individual nodeId to process tags against"
+                        },
+                        "nodes": {
+                            "type": "null"
+                        }
+                    },
+                    "required": ["nodeId"]
+                },
+                {
```

Comment:
How about only use arrays of nodes, don't need support nodeId option. I think for usage perspective, the array of nodes could include one nodeId when array of nodes length is 1, and only contain one nodeId. it will be not complex from usage or codes implementation.